### PR TITLE
:tada::one::one: website_sale_no_sale

### PR DESCRIPTION
**Pull request by trojikman**

### CODE REVIEW

File: website_sale_no_sale/tests/test_no_sale_no_prices.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Copyright 2018 Denis Mudarisov <https://it-projects.info/team/trojikman>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+import odoo.tests
+from odoo import api
+
+
+@odoo.tests.common.at_install(False)
```
Comment (yelizariev) [inline website_sale_no_sale/tests/test_no_sale_no_prices.py]: Set to True and explain why it has to be False

File: website_sale_no_sale/tests/test_no_sale_no_prices.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        url = '/shop/product/ipod-20'
+        code = """
+                    if ($('b[data-oe-expression="product.website_price"]').length > 0) {
+                        console.log('error', 'Price is not removed');
+                    } else {
+                        console.log('ok');
+                    }                   
+        """
+        self.phantom_js(url, code, login="demo")
+        phantom_env.ref('website_sale_no_sale.hide_price').write({'active': False})
```
Comment (yelizariev) [inline website_sale_no_sale/tests/test_no_sale_no_prices.py]: You must check that page is loaded when you check that something is removed

**Comment by yelizariev** [comment]
- [ ] @ufaks make html description for this module

File: website_sale_no_sale/tests/test_no_sale_no_prices.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        self.phantom_js(url, code, login="demo", ready="$('.product_price')")
+
+        if custom_website_view:
+            custom_website_view.write({'active': False})
+        else:
+            phantom_env.ref('website_sale_no_sale.hide_products_prices').write({'active': False})
+
+    def test_02_remove_add_to_cart(self):
+        phantom_env = api.Environment(self.registry.test_cr, self.uid, {})
+        website_id = phantom_env['res.config.settings'].default_get(['website_id'])['website_id']
```
Comment (ufaks) [inline website_sale_no_sale/tests/test_no_sale_no_prices.py]: @trojikman Please use setUp() and tearDown() methods for code duplicates

File: website_sale_add_to_cart/tests/test_add_to_cart.py [hunk 4]
```diff
@@ -1,12 +1,28 @@
...
     def test_add_to_cart(self):
+        phantom_env = api.Environment(self.registry.test_cr, self.uid, {})
+        website_id = phantom_env['res.config.settings'].default_get(['website_id'])['website_id']
+        custom_website_view = phantom_env['ir.ui.view'].search(
+            [('active', '=', False), ('website_id', '=', website_id), ('key', '=', 'website_sale_add_to_cart.product')])
+
+        if custom_website_view:
+            custom_website_view.write({'active': True})
+        else:
+            phantom_env.ref('website_sale_add_to_cart.product').write({'active': True})
```
Comment (yelizariev) [inline website_sale_add_to_cart/tests/test_add_to_cart.py]: Are you sure that this case is used?

File: website_sale_add_to_cart/tests/test_add_to_cart.py [hunk 5]
```diff
@@ -1,12 +1,28 @@
...
+
         # delay is added to be sure that all elements have been rendered properly
         self.phantom_js("/", "odoo.__DEBUG__.services['web_tour.tour'].run('shop_add_to_cart', 1500)",
                         "odoo.__DEBUG__.services['web_tour.tour'].tours.shop_add_to_cart.ready",
                         login='admin')
+
+        if custom_website_view:
+            custom_website_view.write({'active': False})
+        else:
+            phantom_env.ref('website_sale_add_to_cart.product').write({'active': False})
```
Comment (yelizariev) [inline website_sale_add_to_cart/tests/test_add_to_cart.py]: You don't use to rollback your changes -- it's done automatically

File: website_sale_add_to_cart/website_sale_add_to_cart_views.xml [hunk 6]
```diff
@@ -31,7 +31,7 @@
             </xpath>
         </template>
 
-        <template id="product" inherit_id="website_sale.product" customize_show="True" active="True" name="Quick add to Cart (product page)">
+        <template id="product" inherit_id="website_sale.product" customize_show="True" active="False" name="Quick add to Cart (product page)">
```
Comment (yelizariev) [inline website_sale_add_to_cart/website_sale_add_to_cart_views.xml]: No, please. It's the key feature of the module, so it should be available right after installation without extra configuration. Try to fix the test in a different way

File: website_sale_no_sale/README.rst [hunk 7]
```diff
@@ -0,0 +1,47 @@
+.. image:: https://img.shields.io/badge/license-LGPL--3-blue.png
+   :target: https://www.gnu.org/licenses/lgpl
+   :alt: License: LGPL-3
+
+===================
+ Stop Online Sales
```
Comment (yelizariev) [inline website_sale_no_sale/README.rst]: :+1:

File: website_sale_no_sale/__init__.py [hunk 8]
```diff
@@ -0,0 +1,2 @@
+# Copyright 2018 Denis Mudarisov <https://it-projects.info/team/trojikman>
```
Comment (yelizariev) [inline website_sale_no_sale/__init__.py]: We don't use copyright lines for ``__init__.py`` files

File: website_sale_no_sale/__manifest__.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
+# Copyright 2018 Denis Mudarisov <https://it-projects.info/team/trojikman>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+{
+    "name": """Stop Online Sales""",
+    "summary": """Adds options to disable all sales and hide all prices, but keep products visible at website""",
```
Comment (yelizariev) [inline website_sale_no_sale/__manifest__.py]: :+1:

File: website_sale_no_sale/doc/index.rst [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+Installation
+============
+
+* `Install <https://odoo-development.readthedocs.io/en/latest/odoo/usage/install-module.html>`__ this module in a usual way
+
+Configuration
+=============
+
+* Hide prices in on multiple products page:
+    * Using administrator user go to ``[[ Shop ]] >> [[ Customize ]]``
```
Comment (yelizariev) [inline website_sale_no_sale/doc/index.rst]: Fix formatting. ``[[ Shop ]]`` means that there is top menu ``Shop``

https://odoo-development.readthedocs.io/en/latest/dev/docs/rst-requirements.html#references-to-menu

File: website_sale_no_sale/tests/__init__.py [hunk 11]
```diff
@@ -0,0 +1,4 @@
+# Copyright 2018 Denis Mudarisov <https://it-projects.info/team/trojikman>
```
Comment (yelizariev) [inline website_sale_no_sale/tests/__init__.py]: no copyright for init files

File: website_sale_no_sale/tests/test_no_sale_no_prices.py [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+        if self.custom_website_view:
+            self.custom_website_view.write({'active': True})
+        else:
+            self.phantom_env.ref(template_id).write({'active': True})
+
+    def disable_environment(self, template_id):
+        if self.custom_website_view:
+            self.custom_website_view.write({'active': False})
+        else:
+            self.phantom_env.ref(template_id).write({'active': False})
```
Comment (yelizariev) [inline website_sale_no_sale/tests/test_no_sale_no_prices.py]: I think this case might be removed

File: website_sale_no_sale/tests/test_no_sale_no_prices.py [hunk 13]
```diff
@@ -10,37 +10,43 @@
...
-        else:
+    def disable_conflict_module(self, template_id):
+        module, name = template_id.split('.')
+        custom_website_view = self.phantom_env['ir.ui.view'].search(
+            [('active', '=', True), ('website_id', '=', self.website_id), ('key', '=', template_id)])
+
+        if custom_website_view:
+            custom_website_view.write({'active': False})
+        elif self.phantom_env['ir.model.data'].search([('module', '=', module), ('name', '=', name)]):
+            print('**** Statement is TRUE, what the fuck!!!')
```
Comment (ufaks) [inline website_sale_no_sale/tests/test_no_sale_no_prices.py]: @trojikman Maybe it's better to delete this print ;)

File: website_sale_no_sale/tests/test_no_sale_no_prices.py [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+        self.phantom_env = api.Environment(self.registry.test_cr, self.uid, {})
+        self.website_id = self.phantom_env['res.config.settings'].default_get(['website_id'])['website_id']
+        custom_website_view = self.phantom_env['ir.ui.view'].search(
+            [('active', '=', False), ('website_id', '=', self.website_id),
+                ('key', '=', template_id)])
+
+        if custom_website_view:
+            custom_website_view.write({'active': True})
+        else:
+            self.phantom_env.ref(template_id).write({'active': True})
```
Comment (yelizariev) [inline website_sale_no_sale/tests/test_no_sale_no_prices.py]: I think this case might be removed

File: website_sale_no_sale/tests/test_no_sale_no_prices.py [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+
+    def disable_conflict_module(self, template_id):
+        module, name = template_id.split('.')
+        custom_website_view = self.phantom_env['ir.ui.view'].search(
+            [('active', '=', True), ('website_id', '=', self.website_id), ('key', '=', template_id)])
+
+        if custom_website_view:
+            custom_website_view.write({'active': False})
+        elif self.phantom_env['ir.model.data'].search([('module', '=', module), ('name', '=', name)]):
+            print('**** Statement is TRUE, what the fuck!!!')
```
Comment (yelizariev) [inline website_sale_no_sale/tests/test_no_sale_no_prices.py]: don't use print

File: website_sale_no_sale/tests/test_no_sale_no_prices.py [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+            self.phantom_env.ref(template_id).write({'active': False})
```
Comment (yelizariev) [inline website_sale_no_sale/tests/test_no_sale_no_prices.py]: I think this case might be removed

File: website_sale_no_sale/tests/test_no_sale_no_prices.py (hunk 14 above)
Comment (trojikman) [reply website_sale_no_sale/tests/test_no_sale_no_prices.py]: Without this case tests will not work correctly depending on cases when this module installed with website_multi_theme or without it

**Comment by yelizariev** [comment]
:green_heart: is for development only now, not for releasing. Use :rainbow: instead

**Comment by yelizariev** [comment]
@ufaks add Key features block please
https://odoo-development.readthedocs.io/en/latest/description/index.html.html#key-features

* "Hide Add to Cart" option
* "Hide price" option

**Pull request merged**